File: SmartHouseFrontend/src/app/shared/models/SmartProperty.ts
import { TypeOfProperty } from "./TypeOfProperty";
import { Activation } from "./Activation";
import { SmartDeviceDTO } from "./SmartDevice";


export interface SmartPropertyListDTO{
    id:string,
    typeOfProperty: TypeOfProperty,
    name: string,
    city:string,
    country:string,
    address: string,
    isAccepted:Activation
    
}
export interface SmartPropertyWithUserDTO{
    id:string,
    typeOfProperty: TypeOfProperty,
    name: string,
    city:string,
    country:string,
    address: string,
    ownerName?:string,

}
export interface PendingSmartPropertyListDTO{
    id:string,
    typeOfProperty: TypeOfProperty,
    name: string,
    address: string,
    city: string,
    country: string,
    quadrature:number,
    numberOfFloors:number,
    userName:string
}

export interface SinglePropertyDTO{
        typeOfProperty:TypeOfProperty;
        name:string,
        address:string,
        city:string,
        country:string,
        quadrature:number,
        numberOfFloors:number,
        longitude:number,
        latitude:number,
        image:string,
        imageType:string
        devices:SmartDeviceDTO[]
}

export interface SmartPropertyRegisterDTO{
        typeOfProperty:TypeOfProperty;
        name:string,
        address:string,
        city:string,
        country:string,
        quadrature:number,
        numberOfFloors:number,
        longitude:number,
        latitude:number,
        image:string,
        imageType:string
}
export interface ProccesRequest{
    Accept:boolean,
    Reason?:string
}

File: SmartHouseBackend/SmartHouse.Core/Model/SmartProperty.cs
namespace SmartHouse.Core.Model
{
    public enum TypeOfProperty
    {
        House, Apartment
    }
    public enum Activation
    {
        Accepted, Rejected, Pending
    }
    public class SmartProperty
    {
        public Guid Id { get; set; }
        public TypeOfProperty TypeOfProperty { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Address { get; set; } = string.Empty;
        public string City { get; set; } = string.Empty;
        public string Country { get; set; } = string.Empty;

        public double Quadrature { get; set; }
        public Activation IsAccepted { get; set; }
        public int NumberOfFloors { get; set; }

        public double Longitude { get; set; }
        public double Latitude { get; set; }
        public string PathToImage { get; set; } = string.Empty;

        public User User { get; set; }
        public List<SmartDevice> Devices { get; set; } = new();
        public string? Reason { get; set; } = string.Empty;
    }
}
